src/widgets/colorchooser.h

```diff
@@ -0,0 +1,251 @@
+#ifndef COLORCHOOSER_H
+#define COLORCHOOSER_H
+
+#include <QLineEdit>
+#include <QProxyStyle>
+#include <QWidget>
+#include <qapplication.h>
+#include <qgroupbox.h>
+#include <qlabel.h>
+#include <qpushbutton.h>
+#include <qslider.h>
+#include <qstackedwidget.h>
+#include <qdebug.h>
+#include <qpainter.h>
+#include <QSpinBox>
+#include <QDesktopWidget>
+
+
+
+class ColorCircle;
+class CustomStyle1 : public QProxyStyle
+{
+
+public:
+
+	using QProxyStyle::QProxyStyle;
+
+	int styleHint(QStyle::StyleHint hint,
+
+		const QStyleOption* option = 0,
+
+		const QWidget* widget = 0,
+
+		QStyleHintReturn* returnData = 0) const
+	{
+		if (hint == QStyle::SH_Slider_AbsoluteSetButtons) 	return (Qt::LeftButton | Qt::MidButton | Qt::RightButton);
+
+		return QProxyStyle::styleHint(hint, option, widget, returnData);
+	}
+};
+
+
+
+class CustomBackground : public QWidget
+{
+	Q_OBJECT
+public:
+	bool isBig = false;
+
+	CustomBackground(QWidget *parent = Q_NULLPTR, Qt::WindowFlags f = Qt::WindowFlags()) {
+		QWidget::QWidget(parent);
+		setWindowFlags(Qt::FramelessWindowHint | Qt::X11BypassWindowManagerHint | Qt::WindowStaysOnTopHint);
+		setWindowFlag(Qt::SubWindow);
+		setGeometry(QApplication::desktop()->screenGeometry().width() / 2, QApplication::desktop()->screenGeometry().height() / 2, 3, 3);
+	}
+
+	void drawPixmap(QPixmap pm) {
+		setGeometry(0, 0, QApplication::desktop()->screenGeometry().width(), QApplication::desktop()->screenGeometry().height());
+		pixmap = &pm;
+		image = pixmap->toImage();
+		isBig = true;
+		setMouseTracking(true);
+		repaint();
+	}
```

Comment:
Move all these definitions to the appropriate cpp file and take them out of the header